src/wp-admin/includes/image.php

```diff
@@ -376,13 +411,30 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 	return _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id );
```

Comment:
See my related comment: I think we need to add an (optional) `$output_mime_type` parameter to this function in order to reliably support creation of missing sub-sizes, and now also missing MIME types. Maybe even a boolean `$is_primary` parameter to indicate whether to store the sub-size data within the image size metadata (in addition to the image size metadata sources).

The code could then be something like this:
```suggestion
	$image_meta = _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id, $primary_mime_type, true );
	foreach ( $additional_mime_types as $additional_mime_type ) {
		$image_meta = _wp_make_subsizes( $new_sizes, $file, $image_meta, $attachment_id, $additional_mime_type, false );
	}
```

Reply:
this function now includes the mime type and that is used internally to output the correct type - https://github.com/WordPress/wordpress-develop/pull/2393/files?w=1#diff-c8719b0a67432e25fb08dad2ad0d081998f835186937dba96966f592bc9d9706R502